pkg/controller/autoops/reconcile.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package autoops
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	autoopsv1alpha1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/autoops/v1alpha1"
+	esv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/deployment"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/reconciler"
+	ulog "github.com/elastic/cloud-on-k8s/v3/pkg/utils/log"
+)
+
+func (r *ReconcileAutoOpsAgentPolicy) doReconcile(ctx context.Context, policy autoopsv1alpha1.AutoOpsAgentPolicy, state *State) *reconciler.Results {
+	log := ulog.FromContext(ctx)
+	log.V(1).Info("Reconcile AutoOpsAgentPolicy")
+
+	results := reconciler.NewResult(ctx)
+
+	// Enterprise license check
+	enabled, err := r.licenseChecker.EnterpriseFeaturesEnabled(ctx)
+	if err != nil {
+		return results.WithError(err)
+	}
+	if !enabled {
+		msg := "AutoOpsAgentPolicy is an enterprise feature. Enterprise features are disabled"
+		log.Info(msg)
+		state.UpdateInvalidPhaseWithEvent(msg)
+		return results.WithRequeue(5 * time.Minute)
+	}
+
+	// run validation in case the webhook is disabled
+	if err := r.validate(ctx, &policy); err != nil {
+		state.UpdateInvalidPhaseWithEvent(err.Error())
+		return results.WithError(err)
```

Comment:
should we record an event here?

Reply:
Same question as https://github.com/elastic/cloud-on-k8s/pull/8941#discussion_r2604108487